internal/jobservice/application.go

```diff
@@ -30,12 +34,47 @@ func New() *App {
 	return &App{}
 }
 
+var DefaultConfiguration = &configuration.JobServiceConfiguration{
+	GrpcPool: grpcconfig.GrpcPoolConfig{
+		InitialConnections: 5,
+		Capacity:           5,
+	},
+}
+
+// Mutates config where possible to correct mis-configurations.
+// Returns a non-nil error if mis-configuration is unrecoverable.
+func RectifyConfig(config *configuration.JobServiceConfiguration) error {
```

Comment:
```suggestion
func RectifyConfig(config *configuration.JobServiceConfiguration) {
```

I don't think you ever return an error here.

Reply:
I don't, but in the future I imagine there'd be a need to. This is only checking config variables I added in this PR but eventually we should probably be doing some kind of validation of all configuration options.

Reply:
This seems to me to be YAGNI.  If we expand to do validation we could add returning errors but as of right now we don't need the return.  I'll resolve though if you think that best.